lib/pgBackRest/Config/Config.pm

```diff
@@ -2556,40 +448,41 @@ sub optionValidate
                 }
 
                 # If a depend value exists, make sure the option value matches
-                if ($bDependResolved && defined($$oDepend{&OPTION_RULE_DEPEND_VALUE}) &&
-                    $$oDepend{&OPTION_RULE_DEPEND_VALUE} ne $strDependValue)
+                if ($bDependResolved && cfgRuleOptionDependValueTotal($iCommandId, $iOptionId) == 1 &&
+                    cfgRuleOptionDependValue($iCommandId, $iOptionId, 0) ne $strDependValue)
```

Comment:
This is a bit weird but I want to address it when we rewrite this in C.  Right now there can be a depend value or a depend list, but a value is just a list of 1.  That's what this is hard-coded to 0.  I didn't want to create a different C function for something we'll get rid of, but also didn't want to change the Perl code too much so it would be easier to review changes.